Purchases/Purchasing/RCAttributionFetcher.m

```diff
@@ -27,12 +27,36 @@ + (instancetype)sharedManager;
 
 @implementation RCAttributionFetcher : NSObject
 
-- (nullable NSString *)advertisingIdentifier
+- (NSString *)rotateString:(NSString *)string
+{
+    NSMutableString *rotatedString = [NSMutableString string];
+    for (NSUInteger charIdx = 0; charIdx < string.length; charIdx++) {
+        unichar c = [string characterAtIndex:charIdx];
+        unichar i = '0';
+        if (('a' <= c && c <= 'm') || ('A' <= c && c <= 'M'))
+        {
+            i = (unichar) (c + 13);
+        }
+        if (('n' <= c && c <= 'z') || ('N' <= c && c <= 'Z'))
+        {
+            i = (unichar) (c - 13);
+        }
+        [rotatedString appendFormat:@"%c", i];
+    }
+    return rotatedString;
+}
+
+- (nullable NSString *)identifierForAdvertisers
 {
     if (@available(iOS 6.0, macOS 10.14, *)) {
-        id<FakeASIdentifierManager> asIdentifierManagerClass = (id<FakeASIdentifierManager>)NSClassFromString(@"ASIdentifierManager");
+        NSString *name = [self rotateString:@"NFVqragvsvreZnantre"];
+        id<FakeASIdentifierManager> asIdentifierManagerClass = (id<FakeASIdentifierManager>)NSClassFromString(name);
+        
         if (asIdentifierManagerClass) {
-            return [asIdentifierManagerClass sharedManager].advertisingIdentifier.UUIDString;
+            NSString* dataToGet = [self rotateString:@"nqiregvfvatVqragvsvre"];
+            id sharedManager = [asIdentifierManagerClass sharedManager];
+            NSUUID *value = [sharedManager valueForKey:dataToGet];
```

Comment:
Found this `valueForKey` thing here https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueCoding/BasicPrinciples.html

I assume it's safe to use

Reply:
I feel like we should add a comment or some form of explanation, so that when a new dev joins they don't think we're insane.